OpenRA.Mods.Common/Widgets/Logic/MissionBrowserLogic.cs

```diff
@@ -177,15 +178,42 @@ void SelectMap(MapPreview preview)
 			selectedMapPreview = preview;
 
 			// Cache the rules on a background thread to avoid jank
-			new Thread(() => selectedMap.PreloadRules()).Start();
+			var difficultyDisabled = true;
+			var difficulties = new string[0];
+			var briefing = "";
 
-			var briefingVideo = selectedMap.Videos.Briefing;
-			var briefingVideoVisible = briefingVideo != null;
-			var briefingVideoDisabled = !(briefingVideoVisible && modData.DefaultFileSystem.Exists(briefingVideo));
+			var briefingVideo = "";
+			var briefingVideoVisible = false;
+			var briefingVideoDisabled = true;
 
-			var infoVideo = selectedMap.Videos.BackgroundInfo;
-			var infoVideoVisible = infoVideo != null;
-			var infoVideoDisabled = !(infoVideoVisible && modData.DefaultFileSystem.Exists(infoVideo));
+			var infoVideo = "";
+			var infoVideoVisible = false;
+			var infoVideoDisabled = true;
+
+			new Thread(() =>
+			{
+				selectedMap.PreloadRules();
+				var mapOptions = selectedMap.Rules.Actors["world"].TraitInfo<MapOptionsInfo>();
+
+				difficulty = mapOptions.Difficulty;
```

Comment:
It seems that maps have this set to `null` by default, even when they have difficulties defined. So this should probably be `difficulty = mapOptions.Difficulty ?? mapOptions.Difficulties.FirstOrDefault();`